### PR TITLE
Make figure3

### PR DESCRIPTION
**Pull request by murphymadeleine21**
Looks pretty nice. @cyrillustan check out this thread: https://asmlab.slack.com/archives/C01CJV8SH2S/p1643243773015000

### CODE REVIEW

**Review by aarmey — CHANGES_REQUESTED** [review]
Looks like you got an error from the cross-validation splitting in such a way that one class was completely missing in the training set. Are you sure the folds are shuffled and that you're using 10 folds? How many cases are there of severe and deceased? This outcome should be very unlikely unless there are few cases of one class, or the folds aren't shuffled.

File: syserol/COVID.py [hunk 1]
```diff
@@ -104,10 +104,11 @@ def dimensionLabel3D():
...
-    subjj = df["group"].isin(["Severe", "Deceased"])
+    patients = np.unique(df['patient_ID'], return_index=True)
+    subjj = df.iloc[np.sort(patients[1])]['group'].isin(["Severe", "Deceased"])
 
     X = tfac.factors[0][subjj, :]
-    y = pd.factorize(df.loc[subjj, "group"])[0]
+    y = pd.factorize(df.loc[subjj[subjj].index,"group"])[0]
     aucs = []
 
     kf = KFold(n_splits=10, shuffle=True)
```
Comment (murphymadeleine21) [inline syserol/COVID.py:114]: Hm. That's weird. I mean we're very clearly here setting it to a 10 fold split with shuffling.

**Comment by murphymadeleine21** [comment]
There is an inbalance between the number of severe vs. deceased patients, but not enough that we would ever have a training set where there are none of the smaller set. We have 35 Deceased patients and 76 Severe, and the training sets are all 99 or 100 in size.

**Comment by murphymadeleine21** [comment]
Oh, though the error is saying there's only one type present in the y_true, so seems like it's happening in the test set. If the test set is about 10 it could easily all be one type. Maybe need to change the fold number

**Comment by aarmey** [comment]
Try stratifying your folds: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedKFold.html

**Review by aarmey — APPROVED** [review]
Great! Merging.
